arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1304,10 +1304,18 @@ Result RocksDBCollection::updateDocument(transaction::Methods* trx,
   TRI_ASSERT(trx->state()->isRunning());
   TRI_ASSERT(_objectId != 0);
   Result res;
-
+  
   RocksDBMethods* mthds = RocksDBTransactionState::toMethods(trx);
-  // disable indexing in this transaction if we are allowed to
-  IndexingDisabler disabler(mthds, trx->isSingleOperationTransaction());
+
+  // we may only disable indexing if we don't have a unique array index
+  bool mayDisableIndexing = trx->isSingleOperationTransaction();
+  for (std::shared_ptr<Index> const& idx : _indexes) {
```

Comment:
Should do the job, however, isn't it very expensive to always do this check?
A few things to consider here:
* if `ḿayDisableIndexing` is false from the beginning, then there is no need to enumerate all indexes at all
* if `mayDisableIndexing` is set to false in the loop, the loop can be aborted

Apart from that, `_indexes` is accessed without holding the lock.

Reply:
I am wondering if it is possible to turn on indexing just for the specific index, but not for all others.
That would need to happen on index level of course and not here. That would be my preferred solution, because it will not slow anything down.